src/java/src/main/java/com/dnanexus/DXSearch.java

```diff
@@ -308,11 +309,43 @@ FindDataObjectsRequest buildRequestHash(int limit) {
          * @return object encapsulating the result set
          */
         public FindDataObjectsResult<T> execute(int pageSize) {
-            return new FindDataObjectsResult<T>(this.buildRequestHash(pageSize), this.classConstraint,
+            return new FindDataObjectsResult<T>(this.buildRequestHash(pageSize, null), this.classConstraint,
                     this.env, pageSize);
         }
 
         /**
+         * Executes the query and returns first result set page.
+         *
+         * @param pageSize Page size
+         *
+         * @return Iterable result set page
+         */
+        public SearchPage<T> getFirstPage(int pageSize) {
+            if (pageSize <= 0) {
+                throw new IllegalArgumentException("Invalid page size - should be positive");
+            }
+            return new SearchPage<T>(this.buildRequestHash(pageSize, null), this.classConstraint, this.env);
+        }
+
+        /**
+         * Executes the query and returns subsequent result set page.
+         *
+         * @param pageSize Page size
+         * @param starting Result of SearchPage<T>.getNext() call on previous page
+         *
+         * @return Iterable result set page
+         */
+        public SearchPage<T> getSubsequentPage(int pageSize, JsonNode starting) {
+            if (pageSize <= 0) {
+                throw new IllegalArgumentException("Invalid page size - should be positive");
+            }
+            if (starting == null) {
```

Comment:
Typically for errors of this type we would throw NullPointerException. In storozhilov-dnanexus#2 I changed this to use Preconditions.checkNotNull which, in addition to being a bit more compact, also changes the exception type that is thrown.